usersession/userd/privileged_desktop_launcher.go

```diff
@@ -75,6 +82,10 @@ func (s *PrivilegedDesktopLauncher) IntrospectionData() string {
 // DBus interface. The desktopFileID is described here:
 // https://standards.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#desktop-file-id
 func (s *PrivilegedDesktopLauncher) OpenDesktopEntry(desktopFileID string, sender dbus.Sender) *dbus.Error {
+	return s.OpenDesktopEntry2(desktopFileID, "", nil, nil, sender)
+}
+
+func (s *PrivilegedDesktopLauncher) OpenDesktopEntry2(desktopFileID string, action string, uris []string, environment map[string]string, sender dbus.Sender) *dbus.Error {
```

Comment:
Could you add a check that `len(environment) == 0`? Until we've got the code in place to pass startup notification tokens through, we should reject all environment variables.